src/NHibernate/Cfg/XmlHbmBinding/MappingLogExtensions.cs

```diff
@@ -1,22 +1,39 @@
 using System.Linq;
+using NHibernate.Mapping;
+using NHibernate.Type;
 
 
 namespace NHibernate.Cfg.XmlHbmBinding
 {
 	public static class MappingLogExtensions
 	{
-		public static void LogMapped(this Mapping.Property property, IInternalLogger log)
+		public static void LogMapped(this Property property, IInternalLogger log)
 		{
 			if (log.IsDebugEnabled)
 			{
-				string msg = "Mapped property: " + property.Name;
-				string columns = string.Join(",", property.Value.ColumnIterator.Select(c => c.Text).ToArray());
+				var msg = "Mapped property: " + property.Name;
+				var columns = string.Join(",", property.Value.ColumnIterator.Select(c => c.Text).ToArray());
 				if (columns.Length > 0)
 					msg += " -> " + columns;
-				if (property.Type != null)
-					msg += ", type: " + property.Type.Name;
+				var propertyTypeName = SafeGetPropertyTypeName(property);
+				if (propertyTypeName != null)
+					msg += ", type: " + propertyTypeName;
 				log.Debug(msg);
 			}
 		}
+
+		static string SafeGetPropertyTypeName(Property property)
+		{
+			try
+			{
+				//property.Type property can trigger a type load
+				if (property.Type != null) return property.Type.Name;
+			}
+			catch (MappingException)
```

Comment:
At that point it is not yet converted to a `MappingException`. This occurs later (higher in the stack). So here, for our two current failing test cases, it can be an `InvalidOperationException` or an `InstanciationException`.

The more "future proof" here would be to swallow any `Exception`.

I wonder whether we should additionally log that exception or not. Since this occurs when generating debug log only, I would tend to log it as debug too, if we log it at all.